File: Data Structures & Algorithms/binary-tree-preorder-traversal/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        
        def traverse(root, result):
            if not root:
                return None, result
            result.append(root.val)
            traverse(root.left, result)
            traverse(root.right, result)
            return root, result
        
        _, result = traverse(root, [])

        return result

File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-1.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func inorderTraversal(root *TreeNode) []int {

	result := []int{}

	traverse(root, &result)

	return result
}

func traverse(root *TreeNode, result *[]int) {
	if root == nil {
		return 
	}
	fmt.Println(root.Val)
	traverse(root.Left, result)
	*result = append(*result, root.Val)
	traverse(root.Right, result)
}